rnd/autogpt_server/autogpt_server/server/integrations.py

```diff
@@ -84,83 +85,44 @@ async def callback(
     )
 
 
-@integrations_api_router.get(
-    "/{provider}/credentials", response_model=list[CredentialsMetaResponse]
-)
+class CredentialsResponse(CredentialsMetaResponse):
+    user_email: str
+    scopes: list[str] | None
+
+
+@integrations_api_router.get("/{provider}/credentials")
 async def list_credentials(
     provider: Annotated[str, Path(title="The provider to list credentials for")],
-    user_id: Annotated[str, Depends(get_user_id)],
+    auth: Annotated[dict[str, Any], Depends(auth_middleware)],
     store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
 ) -> list[CredentialsMetaResponse]:
-    credentials = store.get_creds_by_provider(user_id, provider)
+    credentials = store.get_creds_by_provider(auth["sub"], provider)
     return [
-        CredentialsMetaResponse(credentials_id=cred.id, credentials_type=cred.type)
+        CredentialsResponse(
+            credentials_id=cred.id,
+            credentials_type=cred.type,
+            user_email=auth["email"],
+            scopes=cred.scopes if isinstance(cred, OAuth2Credentials) else None,
+        )
         for cred in credentials
     ]
 
 
-class CredentialsDetailResponse(BaseModel):
-    id: str
-    provider: str
-    title: str
-    type: Literal["oauth2", "api_key"]
-    access_token: str
-    refresh_token: Optional[str]
-    access_token_expires_at: Optional[int]
-    refresh_token_expires_at: Optional[int]
-    scopes: list[str]
-    metadata: dict
-
-
-@integrations_api_router.get(
-    "/{provider}/credentials/{cred_id}", response_model=CredentialsDetailResponse
-)
+@integrations_api_router.get("/{provider}/credentials/{cred_id}")
 async def get_credential(
     provider: Annotated[str, Path(title="The provider to retrieve credentials for")],
     cred_id: Annotated[str, Path(title="The ID of the credentials to retrieve")],
     user_id: Annotated[str, Depends(get_user_id)],
     store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
-) -> CredentialsDetailResponse:
+) -> Credentials:
     credential = store.get_creds_by_id(user_id, cred_id)
     if not credential:
         raise HTTPException(status_code=404, detail="Credentials not found")
     if credential.provider != provider:
         raise HTTPException(
             status_code=400, detail="Credentials do not match the specified provider"
```

Comment:
Should be 404 imo: within the scope of the given provider, no credentials exist with this ID.